### ISSUE TITLE
Lint for Iterator + Copy

### ISSUE THREAD

**Issue by sinkuu** [Manishearth/rust-clippy#1534]
Implementing `Copy` for iterators makes footguns, and [the standard library settled not to do so](https://github.com/rust-lang/rust/pull/21846#issuecomment-110526401) for now.

```rust
let mut it = /* a Copy + Iterator */;
for x in it { if cond { break; } } // `it` copied here (forgot to `&mut it`)
let x = it.next(); // `it` is in the initial state
```

Clippy should have a lint that warns if "use of the original iterator after copied" is detected.

**Comment by oli-obk** [comment]
The E-easy part of this is a lint triggering on any type that implements `Iterator` and `Copy` at the same time. The E-medium part is a lint triggering only on the duplicate use.

**Comment by foonathan** [comment]
Google led me here, but isn't this implemented already as https://rust-lang.github.io/rust-clippy/stable/index.html?search=copy+iterator#copy_iterator?
